nemo/src/model/rule_model/numeric_literal.rs

```diff
@@ -11,6 +13,19 @@ pub enum NumericLiteral {
     Double(Double),
 }
 
+impl NumericLiteral {
+    /// Get primitive type that fits the literal
+    pub fn primitive_type(&self) -> PrimitiveType {
+        match self {
+            Self::Integer(_) => PrimitiveType::Integer,
+            Self::Double(_) => PrimitiveType::Float64,
+            Self::Decimal(_, 0) => PrimitiveType::Integer,
+            // this is a conservative estimate; if possible we transform decimals into Integers or Float64 but if this does not work, we store it as Any (internally as a String)
```

Comment:
I guess technically we never want to convert decimals into float (at least not implicitly), because floating point arithmetic gives different results than working on decimals.

Reply:
One can certainly discuss this. As of now however, there is the following implementation:
```
impl TryFrom<Constant> for LogicalFloat64 {
    type Error = InvalidRuleTermConversion;

    fn try_from(constant: Constant) -> Result<Self, Self::Error> {
        match constant {
            Constant::NumericLiteral(NumericLiteral::Double(d)) => Ok(d.into()),
            Constant::NumericLiteral(NumericLiteral::Decimal(a, b)) => format!("{a}.{b}")
                .parse()
                .ok()
                .and_then(|d: f64| Double::new(d).map(|d| d.into()).ok())
                .ok_or(InvalidRuleTermConversion::new(
                    constant,
                    PrimitiveType::Float64,
                )),
            Constant::NumericLiteral(NumericLiteral::Integer(a)) => LogicalInteger(a)
                .try_into()
                .map_err(|_err| InvalidRuleTermConversion::new(constant, PrimitiveType::Float64)),
            _ => Err(InvalidRuleTermConversion::new(
                constant,
                PrimitiveType::Float64,
            )),
        }
    }
}
```
But it might be that this never comes into action with the dataflow we have.